Common/Brokerages/KrakenBrokerageModel.cs

```diff
@@ -0,0 +1,179 @@
+/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+
+using System.Collections.Generic;
+using System.Linq;
+using QuantConnect.Benchmarks;
+using QuantConnect.Orders;
+using QuantConnect.Orders.Fees;
+using QuantConnect.Securities;
+using QuantConnect.Util;
+using QuantConnect;
+
+namespace QuantConnect.Brokerages
+{
+    public class KrakenBrokerageModel : DefaultBrokerageModel
+    {
+        /// <summary>
+        /// Gets a map of the default markets to be used for each security type
+        /// </summary>
+        public override IReadOnlyDictionary<SecurityType, string> DefaultMarkets { get; } = GetDefaultMarkets();
+
+        /// <summary>
+        /// Constructor for Kraken brokerage model
+        /// </summary>
+        /// <param name="accountType">Cash or Margin</param>
+        public KrakenBrokerageModel(AccountType accountType = AccountType.Cash) : base(accountType)
+        {
+            
+        }
+        public IReadOnlyDictionary<string, decimal> CoinLeverage { get; } = new Dictionary<string, decimal>
+        {
+            {"BTC", 5}, // only with fiats
+            {"ETH", 5},
+            {"USDT", 2}, // only with fiats
+            {"XMR", 2},
+            {"REP", 2}, // eth available
+            {"XRP", 3},
+            {"BCH", 2},
+            {"XTZ", 2}, // eth available
+            {"LTC", 3},
+            {"ADA", 3}, // eth available
+            {"EOS", 3}, // eth available
+            {"DASH", 3}, 
+            {"TRX", 3}, // eth available
+            {"LINK", 3}, // eth available
+            {"USDC", 3}, // only with fiats
+        };
+
+        private readonly List<string> _fiatsAvailableMargin = new() {"USD", "EUR"};
+        private readonly List<string> _onlyFiatsAvailableMargin = new() {"BTC", "USDT", "USDC"};
+        private readonly List<string> _ethAvailableMargin = new() {"REP", "XTZ", "ADA", "EOS", "TRX", "LINK" };
```

Comment:
nit - let's keep private properties and fields at the top and private methods at the bottom 👍